radix-engine/src/errors.rs

```diff
@@ -191,8 +191,8 @@ pub enum CallFrameError {
     ReadSubstateError(ReadSubstateError),
     WriteSubstateError(WriteSubstateError),
 
-    ScanSubstatesError(CallFrameScanSubstateError),
-    TakeSubstatesError(CallFrameTakeSortedSubstatesError),
+    ScanSubstatesError(CallFrameScanSubstatesError),
+    TakeSubstatesError(CallFrameDrainSubstatesError),
```

Comment:
`TakeSubstates` is still hanging around at various places. Can we do a global find and update?